tests/pyfunc/test_model_export_with_class_and_artifacts.py

```diff
@@ -147,6 +147,32 @@ def test_predict(sk_model, model_input):
         test_predict(sk_model=sklearn_knn_model, model_input=iris_data[0]))
 
 
+@pytest.mark.large
+def test_pyfunc_model_log_load_no_run(sklearn_knn_model, main_scoped_model_class, iris_data):
+    sklearn_artifact_path = "sk_model_no_run"
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(sk_model=sklearn_knn_model, artifact_path=sklearn_artifact_path)
+        sklearn_model_uri = "runs:/{run_id}/{artifact_path}".format(
+            run_id=mlflow.active_run().info.run_id,
+            artifact_path=sklearn_artifact_path)
+
+    def test_predict(sk_model, model_input):
+        return sk_model.predict(model_input) * 2
+
+    pyfunc_artifact_path = "pyfunc_model"
+    mlflow.pyfunc.log_model(artifact_path=pyfunc_artifact_path,
```

Comment:
As a sanity check, can we add a line that confirms `mlflow.active_run() == None` before we call `mlflow.pyfunc.log_model`?